### PR TITLE
Fix topk kernel for large cases

### PR DESCRIPTION
**Pull request by rsuderman**
When doing reductions across long sequence lengths its possible to exceed a 2 GB intermediate buffer. These cases need ensure pointer arithmetic occurs in 32-bits or it will incorrectly offset.

### CODE REVIEW

File: sharktank/hip_kernels/kernels/topk_fp16_ukernel.c [hunk 1]
```diff
@@ -28,8 +28,13 @@ extern "C" __global__ void topk_F16I32(const _Float16 *__restrict__ inputValues,
   uint laneID = threadIdx.x;
 
   int linearIndex = groupID;
-  const _Float16 *batchInput = inputValues + linearIndex * reductionSize;
-  const int32_t *batchIndices = inputIndices + linearIndex * reductionSize;
+
+  int64_t linearIndex64 = linearIndex;
+  int64_t reductionSize64 = reductionSize;
```
[the remark this answers is not in the document]
Comment (rsuderman) [reply sharktank/hip_kernels/kernels/topk_fp16_ukernel.c:33]: Yes. The problem is that for prefill longer context lengths end up exceeding 2 GB for the logits array, so the offset would be blow through. Changing to i64 fixed the issue.

**Review by zeeshanhaque21 — APPROVED** [review]
